File: ECGR 2104/Project 4 Nutrition Tracker/fooditem.cpp
#include "fooditem.h"

FoodItem::FoodItem() {
    name = "";
    calories = 0.0;
    totalFat = 0.0;
    totalSugars = 0.0;
    protein = 0.0;
    sodium = 0.0;
}

FoodItem::FoodItem(std::string name, double calories, double totalFat, double totalSugars, double protein, double sodium) {
    this->name = name;
    this->calories = calories;
    this->totalFat = totalFat;
    this->totalSugars = totalSugars;
    this->protein = protein;
    this->sodium = sodium;
}

// Accessors
std::string FoodItem::getName() const {
    return name;
}

double FoodItem::getCalories() const {
    return calories;
}

double FoodItem::getTotalFat() const {
    return totalFat;
}

double FoodItem::getTotalSugars() const {
    return totalSugars;
}

double FoodItem::getProtein() const {
    return protein;
}

double FoodItem::getSodium() const {
    return sodium;
}

// Mutators
void FoodItem::setName(std::string name) {
    this->name = name;
}

void FoodItem::setCalories(double calories) {
    this->calories = calories;
}

void FoodItem::setTotalFat(double totalFat) {
    this->totalFat = totalFat;
}

void FoodItem::setTotalSugars(double totalSugars) {
    this->totalSugars = totalSugars;
}

void FoodItem::setProtein(double protein) {
    this->protein = protein;
}

void FoodItem::setSodium(double sodium) {
    this->sodium = sodium;
}

// Addition operator overloading
FoodItem FoodItem::operator+(const FoodItem& other) const {
    FoodItem result;
    result.setName(name + " + " + other.name);
    result.setCalories(calories + other.calories);
    result.setTotalFat(totalFat + other.totalFat);
    result.setTotalSugars(totalSugars + other.totalSugars);
    result.setProtein(protein + other.protein);
    result.setSodium(sodium + other.sodium);
    return result;
}

File: ECGR 2104 COMPUTER PROG II/Python Code/Project 4 Nutrition Tracker/FoodItem.py
class FoodItem:
    def __init__(self, name, calories, totalFat, totalSugars, protein, sodium):
        self.name = name
        self.calories = calories
        self.totalFat = totalFat
        self.totalSugars = totalSugars
        self.protein = protein  # Corrected variable name
        self.sodium = sodium
    
    def setName(self, n):
        self.name = n

    def setCalories(self, c):
        self.calories = c
    
    def setTotalFat(self, f):
        self.totalFat = f
    
    def setTotalSugars(self, s):
        self.totalSugars = s
    
    def setProtein(self, p):  # Corrected method name
        self.protein = p
    
    def setSodium(self, s):
        self.sodium = s
    
    def getName(self):
        return self.name
    
    def getCalories(self):
        return self.calories
    
    def getTotalFat(self):
        return self.totalFat
    
    def getTotalSugars(self):
        return self.totalSugars
    
    def getProtein(self):  # Corrected me
        return self.protein
    
    def getSodium(self):
        return self.sodium
    
    def __add__(self, other):
        if isinstance(other, FoodItem):
            # Assuming you want to create a new FoodItem by combining attributes
            combined_name = f"{self.name} + {other.name}"
            combined_calories = self.calories + other.calories
            combined_totalFat = self.totalFat + other.totalFat
            combined_totalSugars = self.totalSugars + other.totalSugars
            combined_protein = self.protein + other.protein
            combined_sodium = self.sodium + other.sodium

            return FoodItem(combined_name, combined_calories, combined_totalFat,
                             combined_totalSugars, combined_protein, combined_sodium)
        else:
            raise TypeError("Unsupported operand type for +")

